packages/js/product-editor/src/components/custom-fields/create-modal/create-modal.tsx

```diff
@@ -236,10 +236,12 @@ export function CreateModal( {
 										'key'
 									) }
 									value={ customField.key }
-									onChange={ changeHandler(
-										customField,
-										'key'
-									) }
+									onChange={ ( value ) =>
+										changeHandler(
+											customField,
+											'key'
+										)( value )
+									}
```

Comment:
I don't think this change is necessary anymore after your `| undefined` addition in the `changeHandler` function.

Reply:
Fixed with 06fb5308c899110d213a379c7f3034037e08c970